CONTRIBUTING.md

```diff
@@ -0,0 +1,365 @@
+   have uncovered a bug in Tokio, creating a new issue in the tokio-rs/tokio
+   issue tracker is the way to report it.
+
+2. By helping to triage the issue: This can be done by providing
+   supporting details (a test case that demonstrates a bug), providing
+   suggestions on how to address the issue, or ensuring that the issue is tagged
+   correctly.
+
+3. By helping to resolve the issue: Typically this is done either in the form of
+   demonstrating that the issue reported is not a problem after all, or more
+   often, by opening a Pull Request that changes some bit of something in
+   Tokio in a concrete and reviewable manner.
+
+**Anybody can participate in any stage of contribution**. We urge you to
+participate in the discussion around bugs and participate in reviewing PRs.
+
+### Asking for General Help
+
+If you have reviewed existing documentation and still have questions or are
+having problems, you can open an issue asking for help.
+
+In exchange for receiving help, we ask that you contribute back a documentation
+PR that helps others avoid the problems that you encountered.
+
+### Submitting a Bug Report
+
+When opening a new issue in the Tokio issue tracker, users will be presented
+with a basic template that should be filled in.
+
+```
+<!--
+Thank you for reporting an issue.
+
+Please fill in as much of the template below as you're able.
+-->
+
+## Version
+
+<!--
+List the versions of all `tokio` crates you are using. The easiest way to get
+this information is using `cargo-tree`.
+
+`cargo install cargo-tree`
+(see install here: https://github.com/sfackler/cargo-tree)
+
+Then:
+
+`cargo tree | grep tokio`
+-->
+
+## Platform
+
+<!---
+Output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)
+-->
+
+## Subcrates
+
+<!--
+If known, please specify the affected Tokio sub crates. Otherwise, delete this
+section.
+-->
+
+## Description
+
+<!--
+
+Enter your issue details below this comment.
+
+One way to structure the description:
+
+<short summary of the bug>
+
+I tried this code:
+
+<code sample that causes the bug>
+
+I expected to see this happen: <explanation>
+
+Instead, this happened: <explanation>
+-->
+```
+
+If you believe that you have uncovered a bug, please fill out this form,
+following the template to the best of your ability. Do not worry if you
+cannot answer every detail, just fill in what you can.
+
+The two most important pieces of information we need in order to properly
+evaluate the report is a description of the behavior you are seeing and a simple
+test case we can use to recreate the problem on our own. If we cannot recreate
+the issue, it becomes impossible for us to fix.
+
+In order to rule out the possibility of bugs introduced by userland code, test
+cases should be limited, as much as possible, to using only Tokio APIs.
+
+See [How to create a Minimal, Complete, and Verifiable example][mcve].
+
+[mcve]: https://stackoverflow.com/help/mcve
```

Comment:
It's not easily linkable, but I try to maintain a [Rust-specific MCVE guide](https://stackoverflow.com/tags/rust/info)  (search for "Producing a Minimal, Complete, Verifiable example for Rust code"). I've also created one for [Diesel](https://stackoverflow.com/tags/rust-diesel/info). These may be useful?

Reply:
Oh, I meant to add that we can create similar on the [Tokio tag](https://stackoverflow.com/tags/rust-tokio/info)

Reply:
It would be nice if it was linkable directly... thoughts?

Reply:
> It would be nice if it was linkable directly... thoughts?

I wish. We've asked in various ways:

- [Support anchor names in posts](https://meta.stackexchange.com/q/37894/281829)
- [Could we add a tag-specific MCVE page?](https://meta.stackoverflow.com/q/366424/155423)

Maybe I should just start my own repo...

Reply:
Happy to take improvements as a follow up PR.